internal/clients/http/sources/sources_client.go

```diff
@@ -135,34 +135,10 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId clients.
 	// Sources API currently does not provide a good server-side filtering.
 	auth, err := filterSourceAuthentications(*resp.JSON200.Data)
 	if err != nil {
-		logger.Warn().Msgf("Sources replied with more than one authentications for source: %s", sourceId)
+		logger.Warn().Msgf("Sources did not return any Provisioning authentication for source: %s", sourceId)
 		return nil, err
 	}
 
-	// Get the resource_id which equals to application_id
-	// and check that application_type_id in /applications/<app_id> equals to provisioning id
-	res, err := c.client.ShowApplicationWithResponse(ctx, *auth.ResourceId, headers.AddSourcesIdentityHeader)
-	if err != nil {
-		return nil, fmt.Errorf("cannot list source authentication: %w", err)
-	}
-
-	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
-	if err != nil {
-		if errors.Is(err, clients.NotFoundErr) {
-			return nil, fmt.Errorf("get source ARN call: %w", http.ApplicationNotFoundErr)
-		}
-		return nil, fmt.Errorf("get source ARN call: %w", err)
-	}
-
-	appTypeId, err := c.GetProvisioningTypeId(ctx)
```

Comment:
This deletes one of the two `GetProvisioningTypeId` calls, I wonder if we could remove also the other one that is in `ListProvisioningSources`. I think that is a different case tho.

Reply:
Yeah, unfortunatelly that one is different case, I was also thinking about it, but no luck (at least no simple way to do it). I'll take a look once we get this PR in :)